### PR TITLE
3.11.98 Release notes and tags

### PR DESCRIPTION
**Pull request by sheriff-rh**
Ships 2019-04-03.

### CODE REVIEW

File: release_notes/ocp_3_11_release_notes.adoc [hunk 1]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+==== Bug Fixes
+
+* Administrative users were not able to access the cluster endpoints because permissions were defined incorrectly. Now, the correct permissions have been defined, and administrative users can use the `_cat` endpoints.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1548640[*BZ#1548640*])
+
+* Image garbage collection failed to remove an image correctly if it has only one tag but more than one repository associated with the image. This has now been resolved and garbage collection completes successfully.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1647348[*BZ#1647348*])
+
+* The `docker` registry Health Check would fail if the bucket was empty on AWS S3 environments, returning a `PathNotFound` message. Now, `PathNotFound` is treated a success and Health Check works as expected for empty buckets.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1655641[*BZ#1655641*])
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: as a success?

File: release_notes/ocp_3_11_release_notes.adoc [hunk 2]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1670473[*BZ#1670473*])
+
+* The playbooks and manual configuration steps to redeploy router certificates were replaced with service serving certificates secret. This would overwrite or miss the router wild certificates secret, causing certificate errors due to incorrect certificates redeployed. Now, the playbooks and manual redeployment steps do not overwrite router certificates secret. The router certificates are redeployed based on the specified sub domain or customer certificates.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1672011[*BZ#1672011*])
+
+* The `ImageStream` used in the `BuildConfig` editor did not have edit properties, causing runtime errors in the `BuildConfig` editor. Now, the editor is initializing tags and objects, even if `ImageStream` in the `BuildConfig` is missing or if the user does not have the correct permissions to use it.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1672904[*BZ#1672904*])
+
+* Master pods did not match time zones with worker nodes, which led to errors in logging timestamps. Now the host's timezone configuration is mounted into the control plane pods.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: Now, the

File: release_notes/ocp_3_11_release_notes.adoc [hunk 3]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1674170[*BZ#1674170*])
+
+* When a cluster was installed, the username in the loopback kubeconfig is the same as the hostname of the master. Now, the variable in the playbook is changed to a different value.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: s/username/user name
s/hostname/host name  
(per IBM style guide)

File: release_notes/ocp_3_11_release_notes.adoc [hunk 4]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+* The Ansible Health Check playbook failed when checking the `curator` status. This occurred because the Health Check assumed `curator` is a `deploymentconfig` instead of a `cronjob`, resulting in a failed check. Now, Health Check properly evaluates for a `cronjob` instead of a `deploymentconfig`.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1676720[*BZ#1676720*])
+
+* Some namespaces would be missing from `oc get projects` if more than 1,000 projects were listed. Now, all items correctly appear when looking at large resource lists.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1677545[*BZ#1677545*])
+
+* High network latency existed between `Kibana` and `Elasticsearch` due to either network issues or under-allocated memory for `Elasticsearch`. As a result, `Kibana` would be unusable because of a gateway timeout. Now, changes have been backported from `Kibana` version 6, which allows modification to the ping timeout. Administrators are not able to override the default `pingTimeout` of 3000ms by setting the `ELASTICSEARCH_REQUESTTIMEOUT` environment variable. `Kibana` is functional until the underlying network issues or under-allocated memory conditions can be resolved.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: s/have been/are backported

File: release_notes/ocp_3_11_release_notes.adoc [hunk 5]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1679613[*BZ#1679613*])
+
+* Previously, the upgrade process for `CRI-O` would attempt to stop `docker` on nodes that had been configured to only run `CRI-O`, resulting in playbook failures. Now, the playbook does not stop `docker` on nodes that are configured only for `CRI-O` operation, ensuring successful upgrades.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1685072[*BZ#1685072*])
+
+* Using `MERGE_JSON_LOG=true` would create fields in the record that would cause syntax violations or create too many fields in `Elasticsearch`, causing severe performance problems. Now, users who experience these problems can tune `fluentd` to accommodate their log record fields without errors or `Elasticsearch` performance degradation.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1685243[*BZ#1685243*])
+
+* The SSL and TLS service uses Diffie-Hellman groups with insufficient strength (a key size less than 2048 bytes). As a result the keys are more vulnerable. Now, the key strength has been increased and certificates are more secure.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: As a result,

File: release_notes/ocp_3_11_release_notes.adoc [hunk 6]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1686590[*BZ#1686590*])
+
+* The files that implemented log rotation functionality were not copied to the correct `fluentd` directory. As a result, logs were not being rotated. Now, the container build has been changed to inspect the `fluentd` gem to find out where to install the files. The files that implement log rotation are copied to the correct directory for `fluentd` usage.
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1686941[*BZ#1686941*])
+
+
+[[ocp-3-11-98-enhancements]]
+==== Enhancements
+
+* The command `oc label --list` has been added, and now shows the resource and name of all the labels.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: s/has been added, and now shows/is now added and shows

File: release_notes/ocp_3_11_release_notes.adoc [hunk 7]
```diff
@@ -2182,3 +2182,109 @@ xref:ocp-311-technology-preview[Technology Preview] and now generally available.
...
+(link:https://bugzilla.redhat.com/show_bug.cgi?id=1268877[*BZ#1268877*])
+
+* This enhancement allows the AWS cloud provider to parse additional endpoint configuration and customization of both core Kubernetes and cluster autoscaler environments. AWS now allows custom and private regions which do not follow the conventions of their public cloud endpoints. {product-title} deployments were limited to the public AWS cloud regions only, and this limited the adoption of the product in these scenarios. Additional configuration elements can be added to the `aws.conf` file and will be honored by {product-title} as well as the `cluster-autoscaler` to ensure the correct cloud endpoints are used to automatically provision EBS volumes, load balancers, and EC2 instances.
```
Comment (ahardin-rh) [inline release_notes/ocp_3_11_release_notes.adoc]: regions, which do not

**Comment by ahardin-rh** [comment]
@sheriff-rh Just a few minor comments from me. Otherwise, looks great!

**Comment by mbearer** [comment]
good job, Andrew!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 3 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Ah, that's news to me. I was following the Linux style syntax. Thanks for that heads up!

**Comment by sheriff-rh** [comment]
Thanks @mbearer  and @ahardin-rh . All syntax/changes made; force pushing now!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 2 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 4 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 1 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed! Yeah that's a weird one... I thought about "treated as successful", but it really does mean "Health Check Success", so I went with your suggestion. Thanks!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 5 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

File: release_notes/ocp_3_11_release_notes.adoc (hunk 7 above)
Comment (sheriff-rh) [reply release_notes/ocp_3_11_release_notes.adoc]: Fixed!

**Comment by sheriff-rh** [comment]
Released today - corrected the date. Merging ASAP.

**Pull request merged**